src/pipx/commands/pin.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from pathlib import Path
+
+from pipx.constants import ExitCode
+from pipx.emojis import hazard
+from pipx.util import PipxError
+from pipx.venv import Venv
+
+logger = logging.getLogger(__name__)
+
+
+def pin(venv_dir: Path, verbose: bool, unpin: bool) -> ExitCode:
+    venv = Venv(venv_dir, verbose=verbose)
+    try:
+        package_metadata = venv.package_metadata[venv.main_package_name]
+        if package_metadata.pinned and not unpin:
+            logger.warning(f"{hazard} Package {package_metadata.package} already pinned")
+        elif not package_metadata.pinned and unpin:
+            logger.warning(f"{hazard} Package {package_metadata.package} not pinned")
```

Comment:
Nit: I'd rather use `emojis.sleep` here and log at `INFO` level, as this shouldn't be "hazardous" in any way?

Reply:
However if we log at `INFO` level, no information will be displayed to the user unless they pass `--verbose`.

Reply:
Ah, you're right. Forgot that we changed the default logging level.